### PR TITLE
SAK-44292: FORUMS: Error unique result while Automatically create multiple topics for groups

### PR DESCRIPTION
**Pull request by FerAlvarez95**

### CODE REVIEW

**Comment by ottenhoff** [comment]
is this solving the root issue or is this masking a deeper issue?

**Comment by jesusmmp** [comment]
> is this solving the root issue or is this masking a deeper issue?

@ottenhoff  This should be an unique value:
https://github.com/sakaiproject/sakai/blob/master/msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java#L1530

Query is:

<![CDATA[select membership.name from org.sakaiproject.component.app.messageforums.dao.hibernate.DBMembershipItemImpl as membership
             where membership.topic.id = :id and membership.type = 3 and membership.permissionLevelName = :permissionLevelName
]]>
https://github.com/sakaiproject/sakai/blob/b44d61c999903bdd6eaa6feb8a81968fe618caa3/msgcntr/messageforums-hbm/src/java/org/sakaiproject/component/app/messageforums/dao/hibernate/DBMembershipItemImpl.hbm.xml#L96

We don't know if this is the right solution or should be changed on the query.

Please, could anyone take a look? Thanks!

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java [hunk 1]
```diff
@@ -1527,6 +1527,7 @@ public String getAllowedGroupForRestrictedTopic(final Long topicId, final String
 				.getNamedQuery("findAllowedGroupInTopic")
 				.setLong("id", topicId)
 				.setString("permissionLevelName", permissionName)
+				.setMaxResults(1)
 				.uniqueResult();
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java:1531]: this should be a list() as more than one group/section should be returned.

**Review by ern — CHANGES_REQUESTED** [review]

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java (hunk 1 above)
Comment (FerAlvarez95) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: Done!

Comment (jesusmmp) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: @ern Is there any problem with the changes?

Comment (ottenhoff) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: Earle has a huge refactoring of msgcntr queued up ... with huge performance gains.

Also, I was never able to replicate the reported bug. I tried several times on my local trunk  .....

Comment (jesusmmp) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: @ottenhoff I uploaded a video with the problem. Have you seeen it?

Comment (ottenhoff) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: Yes, I watched several times and followed the steps several times.

Comment (mateullas) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: We have reproduced this issue in our Sakai 20 in production. We introduced a new ticket (SAK-44462) but @jesusmmp has advised that our new ticket in fact duplicated SAK-44292 Maybe our testing plan in SAK-44462 can help to reproduce the issue

Comment (ottenhoff) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: I was able to replicate today. @ern and I are looking at it

Comment (ern) [reply msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java]: yes looking

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java [hunk 2]
```diff
@@ -130,7 +130,10 @@ private boolean isInstructorForAllowedGroup(Long forumId, boolean isForum) {
         try {
             Site site = siteService.getSite(siteId);
             Set<String> groups = getGroupsWithMember(site, getCurrentUserId());
-            return groups.stream().map(site::getGroup).anyMatch(g -> g.getTitle().equals(groupTitle));
+            for(String group: groupTitle) {
+          	  if(groups.stream().map(site::getGroup).anyMatch(g -> g.getTitle().equals(group)))
+          		  return true;
+            }
```
Comment (ern) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java:136]:
```suggestion
            return groups.stream().map(site::getGroup).anyMatch(g -> groupTitle.contains(g.getTitle()));
```

**Review by ern — APPROVED** [review]

**Pull request merged**